pkg/policy/regoapi/fugue.rego

```diff
@@ -25,7 +25,9 @@ resource_types_v0 := input_resource_types
 resource_types := input_resource_types
 
 resources(resource_type) = ret {
-  ret := __resources_by_type(resource_type)
+  ret := {r.id: r |
+     r := __query({"resource_type": resource_type, "scope": {}})[_]
```

Comment:
Forgive my lack of Rego-knowledge, but what does `[_]` as used here actually do?
If I understand the [Rego docs on this](https://www.openpolicyagent.org/docs/latest/policy-language/#variable-keys) correctly, it creates a list (without keys, e.g. not a map / object) out of whatever `__query` returned before.

However, I'm assuming `__query` also returns a list, so isn't `[_]` a no-op here?

Reply:
Yeah, it's maybe a bit terse.  `[_]` is used for _iteration_ here, to go through each element of the list (in this case is the result of `__query`).  Then this iteration contributes towards the outer [object comprehension](https://www.openpolicyagent.org/docs/latest/policy-language/#comprehensions).  This allows us to go from a list to an object.  I've tried to illustrate this concept here: https://play.openpolicyagent.org/p/e9EtUVrjDX